test/models/portfolio_test.rb

```diff
@@ -40,4 +40,22 @@ class PortfolioTest < ActiveSupport::TestCase
     result = portfolio.cash_balance
     assert_equal 7.0, result
   end
+
+  test "#shares_owned calculates correctly" do
+    portfolio = create(:portfolio)
```

Comment:
In this project we are using FactoryBot so you will need to use factories to create a portfolio, student and stock as well.